lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -3716,7 +3734,14 @@ template <typename ElemTy>
 void BoundInterpreterFunction::fwdBatchedReduceAddInstFloatImpl(
     Value *batch, Value *dest, unsigned_t axis, const ShapeVector &eBatchDims,
     const ShapeVector &eDestDims) {
-  staticAssertFloatingPointType(ElemTy);
+  static_assert(
+      std::is_floating_point<ElemTy>::value ||
+          std::is_same<float16_t,
+                       typename std::remove_cv<ElemTy>::type>::value ||
+          std::is_same<bfloat16_t,
+                       typename std::remove_cv<ElemTy>::type>::value ||
+          std::is_same<int, typename std::remove_cv<ElemTy>::type>::value,
+      "This implementation is for floating-point and int32 values only");
```

Comment:
It's a little funny it's called `FloatImpl`. Maybe we should rename to just `fwdBatchedReduceAddInstImpl`

Reply:
Done.